### ISSUE TITLE
[Bug][Vue]: Uncaught (in promise) TypeError: Cannot read properties of null (reading 'isCE')

### ISSUE THREAD

**Issue by imzisy** [web-infra-dev/rsbuild#2601]
### Version

```sh
System:
    OS: MAC
  Browsers:
    Chrome.
  npmPackages:
    "@rsbuild/core": "^0.7.7",
    "@rsbuild/plugin-vue": "^0.7.7",
```

### Details

Currently I'm using rsbuild with vue 3 plugin and getting this error. 

![Screenshot 2024-06-14 at 3 19 57 PM](https://github.com/web-infra-dev/rsbuild/assets/11715415/bdb43e7f-b707-474f-8f65-e012d9892c52)

and I have found this and I'm not sure if it is relater to the issue. I have disable the splitChunks optimization 
   splitChunks: {
      vue: false,
      router: false,
    },
Which fixes the duplicate loading for vue but I'm still getting the error mentioned. 

Cannot read properties of null (reading 'isCE') 

<img width="556" alt="Screenshot 2024-06-12 at 11 46 27 AM" src="https://github.com/web-infra-dev/rsbuild/assets/11715415/4fe624cb-ad4f-43b2-9244-6bad96b680ce">

### Reproduce link

https://github.com/imzisy/rsbuild-vue-issue/tree/main

### Reproduce Steps

yarn dev for both remote and host try to change any compoent and after that do refresh the browser and you shpuld be able to get the error.

for small project the error happen when you refresh the page but in bigger project the error happen while hot reload.

the error happens from rsbuild 0.7.x

**Comment by elliotwestlake** [comment]
Also seeing the same error with `react`.

**Comment by SoonIter** [comment]
I got it from stackoverflow's reply, which might be a multi-instance problem.

https://stackoverflow.com/questions/72036673/typeerror-cannot-read-properties-of-null-reading-isce-custom-component-li
https://github.com/vuejs/core/issues/4344#issuecomment-899064501

**1. If the multi-instance problem is due to the 3rd dependencies in node_modules**

you can set like this

```typescript
// rsbuild.config.ts
source: {
	alias: {
		'vue': path.resolve(__dirname, 'node_modules', 'vue')
	}
}
```

rspack project might follow below

```typescript
// rspack.config.ts
resolve: {
	alias: {
		'vue': path.resolve(__dirname, 'node_modules', 'vue')
	}
}
```

more information can be found in https://rsdoctor.dev/blog/topic/duplicate-pkg-problem

**2. if the multi-instance problem is due to module federation, i think you might config the related configuration**

https://module-federation.io/configure/shared.html#how-to-use-shared-dependencies

**Comment by imzisy** [comment]
I'm not sure if i'm still getting the same error  

<img width="616" alt="Screenshot 2024-06-15 at 10 41 30 PM" src="https://github.com/web-infra-dev/rsbuild/assets/11715415/85b53ad3-5f0d-4dbc-9c64-2b0bd9c879f9">

**Comment by SoonIter** [comment]
@imzisy I cloned your demo, but I can't reproduce it. 

https://github.com/imzisy/rsbuild-vue-issue/blob/51219e4d9991d51d00baf6dfcfcafc5124f6ed01/provider/rsbuild.config.ts#L49-L57

<img width="400" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/79413249/769f546e-cf7c-4cd2-97b2-5a6b813e1a29">

```typescript
      appendPlugins([
        new ModuleFederationPlugin({
          name: `ASSET_REMOTE`,
          filename: `remoteEntry.js`,
          exposes: {
            "./AppIndex": "./src/views/AppIndex",
            "./store": "./src/store/index",
          },
          shared: {
            // vue: {
            //   singleton: true,
            //   requiredVersion: dependencies.vue,
            // },
            // vuex: {
            //   singleton: true,
            //   requiredVersion: dependencies.vuex,
            // },
          },
        }),
      ]);
```

<img width="400" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/79413249/25aec2a9-63b4-46e5-8815-1dfcb81c6068">

I think the reason for this problem is that **your provider and consumer share the same vue instance, but the semantic version span is too large**. You can try to make it smaller. 
:D Moreover, this problem has nothing to do with Rsbuild. I think it is related to module federation.  If it is okay, I will transfer the issue to the mf repository.

**Comment by imzisy** [comment]
@SoonIter  if it is too large why it is breaking ? any idea ?

**Comment by SoonIter** [comment]
> @SoonIter if it is too large why it is breaking ? any idea ?

try to use  the same vue version 

```json
"dependencies": {
	 "vue": "3.2.31",
	 "vuex": "4.0.2"
}
```

<img width="400" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/79413249/769f546e-cf7c-4cd2-97b2-5a6b813e1a29">

**Comment by imzisy** [comment]
> > @SoonIter if it is too large why it is breaking ? any idea ?
> 
> try to use the same vue version
> 
> ```json
> "dependencies": {
> 	 "vue": "3.2.31",
> 	 "vuex": "4.0.2"
> }
> ```
> 
> <img alt="image" width="400" src="https://private-user-images.githubusercontent.com/79413249/340005387-769f546e-cf7c-4cd2-97b2-5a6b813e1a29.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0Njg4NDAsIm5iZiI6MTcxODQ2ODU0MCwicGF0aCI6Ii83OTQxMzI0OS8zNDAwMDUzODctNzY5ZjU0NmUtY2Y3Yy00Y2QyLTk3YjItNWE2YjgxM2UxYTI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE1VDE2MjIyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzYjJkM2UwYWQ1NWNlOGY5YzJlZDNjNzdjMjU0OTg2ZWY2NTdlYThkYzYzN2ZkZGQxZjMzOTk5Nzg3MWE1NDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vIOPiv8AFZkmoQabHDsoLnb-tLTimjegI3IU0YsYsU8">

I tried using the same version, but it still didn't work. Afrer I also changed a component and manually refreshed the page I have seen the same error .

**Comment by elliotwestlake** [comment]
Seems there is already an issue that looks similar, I have posted my reproduction in there

https://github.com/module-federation/core/issues/2617

**Comment by mrcwl** [comment]
When I use rsbuild v0.6.15 everything works fine.
I'm not sure if it's a issues with rsbuild or module federation

**Comment by imzisy** [comment]
now I tried using 0.7.9 looks ok, I hope someone can explain to us how this was fixed.

**Comment by chenjiahan** [comment]
Rspack fixed some MF issues in 0.7.4 and 0.7.2, which may be related to the current issue:

- https://github.com/web-infra-dev/rspack/issues/6740
- https://github.com/web-infra-dev/rspack/pull/6796

For anyone who encounters MF issues, first upgrade to the latest version of Rsbuild (currently v0.7.9)
